spyder/app/mainwindow.py

```diff
@@ -1617,6 +1619,14 @@ def update_lsp_logs(self):
     def post_visible_setup(self):
         """Actions to be performed only after the main window's `show` method
         was triggered"""
+        # Show plugins that were visible
+        for plugin_name, plugin in self._DOCKABLE_WIDGETS.items():
+            plugin.change_visibility(plugin.get_conf_option("visible"))
+            if plugin.get_conf_option("visible"):
+                plugin.dockwidget.show()
+            else:
+                plugin.dockwidget.hide()

```

Comment:
Could you explain what things changed in the new API that make this necessary now? I just want to understand why this was not necessary in Spyder 4.

Reply:
I am not sure if the WindowState that gets saved actually saves this, but I was not able to get the original state to work unless I did this. So I am not sure. This also relates to work on https://github.com/spyder-ide/spyder/pull/13070. When that is in place this will not be needed.

Reply:
Ok, that's really good too know. Thanks!

Then please let me know when this is ready to be merged.